Source/Core/VideoBackends/OGL/TextureCache.cpp

```diff
@@ -32,12 +33,24 @@ namespace OGL
 {
 static u32 s_ColorCbufid;
 static u32 s_DepthCbufid;
-static SHADER s_palette_pixel_shader[3];
+
+struct PaletteShader
+{
+  SHADER shader;
+  GLuint buffer_offset_uniform;
+  GLuint multiplier_uniform;
+  GLuint copy_position_uniform;
+};
+
+static PaletteShader s_palette_shader[3];
 static std::unique_ptr<StreamBuffer> s_palette_stream_buffer;
-static GLuint s_palette_resolv_texture;
-static GLuint s_palette_buffer_offset_uniform[3];
-static GLuint s_palette_multiplier_uniform[3];
-static GLuint s_palette_copy_position_uniform[3];
+static GLuint s_palette_resolv_texture = 0;
+
+static PaletteShader& RefPaletteShader(TLUTFormat tlutfmt)
+{
+  _assert_(IsValidTLUTFormat(tlutfmt));
```

Comment:
I wrote RefPaletteShader to avoid having code littered with `static_cast`, but, now that I think of it, that purpose is served by the new CompilePaletteShader function.